src/main/java/at/hannibal2/skyhanni/features/mining/glacitemineshaft/CorpseLocator.kt

```diff
@@ -0,0 +1,141 @@
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+// TODO: Maybe implement automatic warp-in for chosen players if the user is not in a party.
+object CorpseLocator {
+    private val config get() = SkyHanniMod.feature.mining.mineshaftWaypoints.corpseLocator
+
+    /**
+     * I have no clue what to do for the regex anymore, I'm slowly losing it
+     * REGEX-TEST: §9Party §8> §b[MVP§0+]§b nobaboy§f: §rx: -164, y: 8, z: -154 | (Lapis Corpse)
+     * REGEX-TEST: Guild > §r§6♣ §b[MVP§0+§b] nobaboy§f: x: 141, y: 14, z: -131
+     * REGEX-TEST: §8[§r§6407§r§8] §r§6♣ §b[MVP§0+§b] nobaboy§f: x: -9, y: 135, z: 20 | (Tungsten Corpse)
+     */
+    private val corpseCoordsWaypoint by RepoPattern.pattern(
+        "corpse.waypoints",
+        "^(?:.+?[^x:y0-9] )?(?<playerName>.+)§f: (?:§r)?x: (?<x>-?\\d+),? y: (?<y>-?\\d+),? z: (?<z>-?\\d+)(?:.+)?"
+    )
+
+    private val parsedLocations: MutableList<LorenzVec> = mutableListOf()
+
+    private fun findCorpse() {
+        Minecraft.getMinecraft().theWorld ?: return
+
+        EntityUtils.getAllEntities().filterIsInstance<EntityArmorStand>()
+            .filterNot { corpse ->
+                MineshaftWaypoints.waypoints.any { it.location.distance(corpse.getLorenzVec()) <= 3 }
+            }
+            .filter { entity ->
+                entity.showArms && entity.hasNoBasePlate() && !entity.isInvisible
+            }
+            .forEach { entity ->
+                val helmetDisplayName = StringUtils.stripControlCodes(entity.getCurrentArmor(3).displayName)
+                val corpseType = WaypointsType.entries.firstOrNull { it.helmetName == helmetDisplayName }
+                corpseType?.let { type ->
+                    for (offset in -1..3) {
+                        val canSee = entity.getLorenzVec().add(y = offset).canBeSeen()
+                        if (canSee) {
+                            MineshaftWaypoints.waypoints.add(Waypoint(waypointType = type, location = entity.getLorenzVec().add(y = 1), shared = false))
+                            ChatUtils.chat("Located a ${type.displayText} and marked its location with a waypoint.")
+                            break
+                        }
+                    }
+                }
+            }
+    }
+
+    private fun shareCorpse() {
+        MineshaftWaypoints.waypoints.filterNot { it.shared }
+            .filterNot { corpse ->
+                parsedLocations.any { corpse.location.distance(it) <= 5 }
+            }
+            .filter { it.location.distanceToPlayer() <= 5 }
+            .forEach { corpse ->
+                val location = corpse.location
+                val x = location.x.toInt()
+                val y = location.y.toInt()
+                val z = location.z.toInt()
+                val type = corpse.waypointType.displayText
+
+                HypixelCommands.partyChat("x: $x, y: $y, z: $z | ($type)")
+                corpse.shared = true
+            }
+    }
+
+    @Suppress("UNUSED_PARAMETER")
```

Comment:
do not use UNUSED_PARAMETER suppression. we would need to add this annotation to every event. the correct solution would be to change the intellij settings somehwere